src/registry.rs

```diff
@@ -148,3 +166,17 @@ impl Registry {
 		iter.into_iter().map(|i| i.into_compact(self)).collect::<Vec<_>>()
 	}
 }
+
+/// A read-only registry, to be used for decoding/deserializing
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Decode)]
+pub struct RegistryReadOnly {
+	types: Vec<Type<OwnedForm>>,
+}
+
+impl RegistryReadOnly {
+	/// Returns the type definition for the given identifier, `None` if no type found for that ID.
+	pub fn resolve(&self, id: NonZeroU32) -> Option<&Type<OwnedForm>> {
+		self.types.get((id.get() - 1) as usize);
+		None
```

Comment:
AS far as I see this always returns `None`. Is that correct?

Reply:
I see you spotted my deliberate mistake :sweat_smile: